lib/symbValue.ml

```diff
@@ -114,16 +115,21 @@ module Make(Cst:Constant.S) = struct
 
   let bit_at k = function
     | Val (Concrete v) -> Val (Concrete (Cst.Scalar.bit_at k v))
-    | Val (ConcreteVector _|Symbolic _|Label _|Tag _|PteVal _ as x) ->
+    | Val (ConcreteVector _|Symbolic _|Label _|Tag _|PteVal _|Instruction _ as x) ->
         Warn.user_error "Illegal operation on %s" (Cst.pp_v x)
     | Var _ -> raise Undetermined
 
-  let unop op_op op v1 = match v1 with
+  let unop op_op op v1 = 
+  match v1 with
     | Val (Concrete i1) ->
         Val (Concrete (op i1))
     | Val (ConcreteVector _|Symbolic _|Label _|Tag _|PteVal _ as x) ->
         Warn.user_error "Illegal operation %s on %s"
           (Op.pp_op1 true op_op) (Cst.pp_v x)
+    | Val (Instruction _ as x) -> 
+      Warn.warn_always "FIXME: operation %s on %s suspicious with -variant self"
+          (Op.pp_op1 true op_op) (Cst.pp_v x) ;
+      v1
```

Comment:
The warning is a good idea, as there is a difficulty here: some operation on Instructions may be legal for some architectures.

A possibe approach would be to limit operation to the useful ones (here mask32) and to check that the arch is AArch64 as is done for masking of PTE entries (see `orop` in the same files for an example).

However, as a PR to come will change the handling of arch dependent-operation (see https://github.com/herd/herdtools7/blob/f1d9f77da969f83952d1d9b35296c93bf52e173b/lib/symbValue.ml#L291 is you are curious) please do not change anything.